woocommerce/woocommerce-gutenberg-products-block
# [Store Customization] AI-generated store name

## What

This PR adds a new endpoint `/ai/store-title` to update the store title during the store customization process if it is empty.

## Why

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11845


## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

### Setup

Please test the feature on a WooExpress install instead of Jurassic Ninja, as the latter can be very limited in terms of server resources and unstable, as other installs can interfere with your own.

- If you already have a pre-configured install, remove the pre-existing WooCommerce Blocks Plugin and upload the one from this zip file: https://wcblocks.wpcomstaging.com/wp-content/uploads/woocommerce-gutenberg-products-block-11878.zip
- Alternatively, create a new WooExpress install from scratch 
- Remove the pre-existing WooCommerce plugin via SSH and
- Install and activate WooCommerce from the following zip file: [woocommerce.zip](https://github.com/woocommerce/woocommerce-blocks/files/13437958/woocommerce.zip)
- Install and activate WooCommerce Blocks from the following zip file: https://wcblocks.wpcomstaging.com/wp-content/uploads/woocommerce-gutenberg-products-block-11878.zip
- Install and activate WooCommerce Beta Tester (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Go to Settings > General, empty the Site Title and save.
- Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

<img width="1233" alt="Screenshot 2023-10-22 at 10 32 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/816867c6-34e5-40a1-a87d-f4a7cda46429">

### Test the CYS experience

0. Make sure you don't have any products in your store.
1. Head over to Home > Customize your Store:
 
<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/b5d34131-3edc-47ed-a8e4-8e2775ca1bbd">

2. Click on design with AI.
3. Type a description for your business.
4. Finish the customization process and make sure the Site Title shown in the assembler is not empty and it's related to your store business description.
5. Go back to the WooCommerce home and start the process again, make sure the title is not modified this time (because it was not empty from the previous execution).


## Screenshots or screencast




## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog
